thrift/src/main/java/com/linecorp/armeria/internal/thrift/ThriftFunction.java

```diff
@@ -52,27 +52,41 @@
     private final Class<?> serviceType;
     private final String name;
     @Nullable
+    private final Object implementation;
+    @Nullable
     private final TBase<?, ?> result;
     private final TFieldIdEnum[] argFields;
     @Nullable
     private final TFieldIdEnum successField;
     private final Map<Class<Throwable>, TFieldIdEnum> exceptionFields;
     private final Class<?>[] declaredExceptions;
 
+    ThriftFunction(Class<?> serviceType, ProcessFunction<?, ?> func,
+                   @Nullable Object implementation) throws Exception {
+        this(serviceType, func.getMethodName(), func, Type.SYNC,
+             getArgFields(func), getResult(func), getDeclaredExceptions(func), implementation);
+    }
+
     ThriftFunction(Class<?> serviceType, ProcessFunction<?, ?> func) throws Exception {
         this(serviceType, func.getMethodName(), func, Type.SYNC,
-             getArgFields(func), getResult(func), getDeclaredExceptions(func));
+             getArgFields(func), getResult(func), getDeclaredExceptions(func), null);
     }
 
     ThriftFunction(Class<?> serviceType, AsyncProcessFunction<?, ?, ?> func) throws Exception {
         this(serviceType, func.getMethodName(), func, Type.ASYNC,
-             getArgFields(func), getResult(func), getDeclaredExceptions(func));
+             getArgFields(func), getResult(func), getDeclaredExceptions(func), null);
+    }
+
+    ThriftFunction(Class<?> serviceType, AsyncProcessFunction<?, ?, ?> func,
+                   @Nullable Object implementation) throws Exception {
```

Comment:
Could you reorder the constructors a little bit, so that it's like:

- the one that accepts `ProcessFunction` and no implementation
- the one that accepts `ProcessFunction` and implementation
- the one that accepts `AsyncProcessFunction` and no implementation
- the one that accepts `AsyncProcessFunction` and implementation

.. for consistency?

Reply:
Done